bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -610,6 +613,18 @@ private PageDTO createPageBean(String sitemapName, @Nullable String title, @Null
         return bean;
     }
 
+    public static @Nullable String convertItemValueColor(@Nullable String color, @Nullable Item item) {
+        if ("itemValue".equals(color)) {
+            String itemRGBHexCode = null;
+            if (item instanceof ColorItem && item.getState() instanceof HSBType) {
+                HSBType hsbState = (HSBType) item.getState();
+                itemRGBHexCode = "#" + Integer.toHexString(hsbState.getRGB()).substring(2);
+            }
```

Comment:
```suggestion
            if (item.getState() instanceof HSBType hsbState) {
                itemRGBHexCode = "#" + Integer.toHexString(hsbState.getRGB()).substring(2);
            }
```

If the state is of `HSBType` it must be a `ColorItem´. You can further reduce code by using pattern matching.

Reply:
Your proposal will lead to a NPE when item is null. 
But I could have a State as parameter rather than an Item if we consider that a test on HSBType is sufficient.

Reply:
`item != null` would then be sufficient. But does it make sense to call the method at all if the item is null?

Reply:
Changed. I am now passing as argument the item state.